codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/CborSerializerGenerator.kt

```diff
@@ -421,7 +421,7 @@
         rust(
             """
             encoder.map(
-                (${context.valueExpression.asRef()}).len().try_into().expect("`usize` to `u64` conversion failed")
+                (${context.valueExpression.asValue()}).len().try_into().expect("`usize` to `u64` conversion failed")
```

Comment:
Same. Here we should be handling a `&HashMap<String, T>`, so `asRef()` should just print the variable name as is.